test/lit/passes/roundtrip-gc.wast

```diff
@@ -8,33 +8,38 @@
  ;; NOMNL:      (export "export" (func $test))
  (export "export" (func $test))
  ;; CHECK:      (func $test
+ ;; CHECK-NEXT:  (local $0 (ref $\7bi32\7d))
  ;; CHECK-NEXT:  (call $help
- ;; CHECK-NEXT:   (struct.new_default $\7bi32\7d)
- ;; CHECK-NEXT:   (block $label$1 (result i32)
+ ;; CHECK-NEXT:   (block (result (ref $\7bi32\7d))
+ ;; CHECK-NEXT:    (local.set $0
+ ;; CHECK-NEXT:     (struct.new_default $\7bi32\7d)
+ ;; CHECK-NEXT:    )
  ;; CHECK-NEXT:    (nop)
- ;; CHECK-NEXT:    (i32.const 1)
+ ;; CHECK-NEXT:    (local.get $0)
  ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:   (i32.const 1)
  ;; CHECK-NEXT:  )
  ;; CHECK-NEXT: )
  ;; NOMNL:      (func $test (type $none_=>_none)
+ ;; NOMNL-NEXT:  (local $0 (ref $\7bi32\7d))
  ;; NOMNL-NEXT:  (call $help
- ;; NOMNL-NEXT:   (struct.new_default $\7bi32\7d)
- ;; NOMNL-NEXT:   (block $label$1 (result i32)
+ ;; NOMNL-NEXT:   (block (result (ref $\7bi32\7d))
+ ;; NOMNL-NEXT:    (local.set $0
+ ;; NOMNL-NEXT:     (struct.new_default $\7bi32\7d)
+ ;; NOMNL-NEXT:    )
  ;; NOMNL-NEXT:    (nop)
- ;; NOMNL-NEXT:    (i32.const 1)
+ ;; NOMNL-NEXT:    (local.get $0)
```

Comment:
Is it expected that the code got worse after we re-enabled this optimization?

Reply:
Yes, this is kind of a corner case. Before we disabled a bunch of stack IR opts to avoid such situations. Now we no longer disable them, which helps normally, but in corner cases like this, we end up with needing a local due to stack code. This doesn't matter in practice except for roundtripping a stacky binary without opts. This is not new to non-nullable locals - it has been the case for a long time in general, but now we just treat non-nullable locals like other locals.

Reply:
Makes sense, thanks!